src/vm/ecalllist.h

```diff
@@ -1603,10 +1603,10 @@ FCFuncStart(gInteropMarshalFuncs)
     FCFuncElement("DestroyStructure", MarshalNative::DestroyStructure)
     FCFuncElement("UnsafeAddrOfPinnedArrayElement", MarshalNative::FCUnsafeAddrOfPinnedArrayElement)
     FCFuncElement("GetExceptionCode", ExceptionNative::GetExceptionCode)
+    QCFuncElement("GetHINSTANCE", COMModule::GetHINSTANCE)
```

Comment:
According to https://github.com/mono/mono/blob/master/mono/metadata/icall.c#L5739, mono returns -1 in non-Windows platforms (and I've confirmed that behavior). We should do the same. I took a quick look at our own code and I suspect we'll return the base address correctly in non-windows as well (but I haven't confirmed that theory - Faizur should be in a good place to test it). Assuming that's the case, we can special case it.